internal/util/cephconf.go

```diff
@@ -71,7 +76,11 @@ if any ceph commands fails it will log below error message
 */
 // createKeyRingFile creates the keyring files to fix above error message logging.
 func createKeyRingFile() error {
-	_, err := os.Create(keyRing)
+	var err error
+	// create keyring file if it does not exist to support backward compatibility
+	if _, err = os.Stat(keyRing); os.IsNotExist(err) {
+		_, err = os.Create(keyRing)
```

Comment:
I am not sure this  works for directories where a ConfigMap is mounted? Should the mapped ConfigMap not be read-only?

Reply:
No this will be only to backward compatibility or else if the configmap contains only ceph.conf and even if the volume is mounted with readOnly:false it fails here.